spec/person_spec.rb

```diff
@@ -0,0 +1,46 @@
+require_relative '../classes/person'
+
+describe 'Persons' do
+  context 'persons' do
+    it 'should initialize a person' do
+      person = Person.new(20, 'Precious', parent_permission: false)
+      expect(person.id).to be_kind_of Numeric
+      expect(person.name).to eql 'Precious'
+      expect(person.parent_permission).to eql false
+      expect(person.age).to eql 20
+      expect(person.rentals).to eql []
+    end
+
+    it 'should add rentals' do
+      person = Person.new(20, 'Precious', parent_permission: false)
+
+      rental1 = double('rental')
+      rental2 = double('rental')
+      allow(rental1).to receive(:person=)
+      allow(rental2).to receive(:person=)
+
+      person.add_rental(rental1)
+      person.add_rental(rental2)
+
+      expect(person.rentals.length).to eql 2
+    end
+
+    it 'should not allow a person to use services' do
+      person = Person.new(17, 'Precious', parent_permission: false)
+
+      expect(person.can_use_services?).to be false
+    end
+
+    it 'should allow a person to use services' do
+      person = Person.new(18, 'Precious', parent_permission: false)
+
+      expect(person.can_use_services?).to be true
+    end
+
+    it 'should correct a person\'s name' do
+      person = Person.new(18, 'Precious', parent_permission: false)
+
+      expect(person.name).to eql 'Precious'
+    end
```

Comment:
- [ ] Great job so far 👏🏻 👏🏻 However, according to the project requirements, there should be unit tests for all the methods in all the classes, Kindly make sure the `of_age?`, and  `correct_name` class person method has (at least) an 'it' block 👇🏻 check your person.rb file



```ruby 

 def correct_name
    @name
  end

  private

  def of_age?
    @age >= 18
  end
```

Reply:
@BenMukebo hello ✋🏻 
We have "it" block for `age ` and for `correct-name` , please check `it 'should correct a person\'s name' do` and `it 'should allow a person to use services' do` .
Thank you, we're waiting for your response

Reply:
@BenMukebo, thanks for the review. However, according to best practices, it is advisable to test only public methods hence, the exclusion of `of_age?`.

Reply:
@Hisoka37, Sorry fo the late reply, I had an emergency  but I'm happy to see that the next code reviewer help you approving this PR, About the the `of_age` it was one of the project requiemrnt to the all the method without specifying (publick and private) but you're also right it is advisable to test only public methods hence, the exclusion of of_age?.